lib/connextclient/ConnextClient.ts

```diff
@@ -893,7 +906,11 @@ class ConnextClient extends SwapClient {
       }
 
       this.logger.trace(`sending request to ${endpoint}${payloadStr ? `: ${payloadStr}` : ''}`);
+
+      const reqId = Date.now() + Math.random();
```

Comment:
I think rather than using a `Map` here with this unique-ish id, we should just be able to use a `Set` of the pending requests, and then when the request is over do `this.pendingrequests.delete(req)`.

Reply:
Good idea.